Add de.klayout.KLayout

willsALMANJ opened the pull request:
This PR adds packaging for KLayout (https://www.klayout.de/, https://github.com/KLayout/klayout). The maintainer has been asked a couple times about packaging as a Flatpak but seems uninterested (https://github.com/KLayout/klayout/issues/125).

LeandroStanger commented on de.klayout.KLayout.yaml:
```diff
@@ -0,0 +1,41 @@
+    url: https://www.klayout.org/downloads/source/klayout-0.26.5.tar.gz
+    sha256: 3f58be531b5570e9d01ac8077a84b90b91fe181e01dd4acc505c8a77dcdc8d88
+  - type: file
+    path: de.klayout.KLayout.appdata.xml
```
New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

willsALMANJ replied:
Okay, I switched out the appdata file for a metainfo file.

LeandroStanger commented:
bot, build de.klayout.KLayout

willsALMANJ commented:
Hmm, looking at the build pipeline -- I don't know that anyone has ever tried to build KLayout for ARM, but I guess it is okay if the build is successful.

bilelmoussaoui commented:
How did you ended up choosing de.klayout.Klayout as an app-id?

bilelmoussaoui commented on de.klayout.KLayout.metainfo.xml:
```diff
@@ -0,0 +1,32 @@
+    </p>
+  </description>
+  
+  <launchable type="desktop-id">org.zotero.KLayout.desktop</launchable>
```
That doesn't look right?

willsALMANJ replied:
Oops, I copied that over from the appdata file which I modeled on the Zotero flatpak. I will fix it after we resolve the app ID question.

willsALMANJ commented:
I went with de.klayout.KLayout because the KLayout project owns the klayout.de web domain, but I can change it to something else if that is better.

bilelmoussaoui commented:
I just noticed that the mentioned domain name on the github repository https://github.com/KLayout/klayout is klayout.org, but it seems it just redirects to the klayout.de.

bilelmoussaoui commented on de.klayout.KLayout.yaml:
```diff
@@ -0,0 +1,41 @@
+  build-commands:
+  - bash ./build.sh -bin ${FLATPAK_DEST}/bin -j$FLATPAK_BUILDER_N_JOBS
+  post-install:
+  - install -Dm644 de.klayout.KLayout.metainfo.xml ${FLATPAK_DEST}/share/appdata/${FLATPAK_ID}.metainfo.xml
```
```suggestion
  - install -Dm644 de.klayout.KLayout.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
```
appdata is the old install directory

bilelmoussaoui commented on de.klayout.KLayout.metainfo.xml:
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
The metainfo should be contributed upstream if possible

willsALMANJ replied:
Okay, I submitted it here: https://github.com/KLayout/klayout/pull/562

willsALMANJ commented:
I think I addressed the feedback. I had a couple questions about the packaging:

* How does flatpak determine the version of the application? I didn't specify it any where. Is it pulling it from the tarball file name?
* KLayout can use Ruby for scripting, so I bundled in Ruby "by hand." Is that the best way to include Ruby? I didn't see a shared module or extension for Ruby support. Or should I have specified org.kde.Sdk as the runtime since that includes Ruby?

nedrichards commented:
@willsALMANJ version as displayed to the user comes from the appdata version tags. Flatpak itself uses ostree to look at the difference between commits - you can have a read about that here: https://docs.flatpak.org/en/latest/under-the-hood.html

Best to just include Ruby rather than making the user get the SDK as well unless you generally need a bunch of developer tools.

barthalion commented on de.klayout.KLayout.yaml:
```diff
@@ -0,0 +1,41 @@
+  - --enable-libgdbm-compat
+  sources:
+  - type: archive
+    url: ftp://ftp.gnu.org/gnu/gdbm/gdbm-1.18.1.tar.gz
```
Can you switch it to https?

willsALMANJ replied:
I switched it. The gdbm site doesn't have an https link, but switching to https worked.

barthalion commented on de.klayout.KLayout.yaml:
```diff
@@ -0,0 +1,41 @@
+app-id: de.klayout.KLayout
```
Please reformat file as described in https://github.com/flathub/flathub/wiki/YAML-Style-Guide

willsALMANJ replied:
The file has been reformatted.

willsALMANJ commented:
bot, build de.klayout.KLayout

willsALMANJ commented:
bot, build de.klayout.KLayout

willsALMANJ commented:
@nedrichards In the metainfo.xml file that I am including (and previously in the appdata file), there is a "versions" section but not a "this version" section, so is it defaulting to using the latest version? Or am I missing somewhere else where it looks for the version?

barthalion commented:
Yes, it uses the latest version specified there.

willsALMANJ commented:
What is the best way to resolve the failure? I switched everything from "appdata" to "metainfo" following [this comment](https://github.com/flathub/flathub/pull/1535#discussion_r428932095) above . Was I supposed to use the metainfo format rules but leave things named "appdata" or something else?

barthalion commented:
The path is fine and the error is legit. It needs to provide `url` and `content_rating` tags.

willsALMANJ commented:
bot, build de.klayout.KLayout

barthalion commented:
/merge @klayoutmatthias
